ducktape/tests/loader.py

```diff
@@ -55,10 +55,13 @@ class TestLoader(object):
 
     def __init__(self, session_context):
         self.session_context = session_context
-        self.logger = session_context.logger
         self.test_file_pattern = DEFAULT_TEST_FILE_PATTERN
         self.test_function_pattern = DEFAULT_TEST_FUNCTION_PATTERN
 
+    @property
```

Comment:
I realized it was potentially unsafe to assume those fields never changed.

For example, the main command-line logic instantiates the session context with a 'None' cluster and then instantiates the cluster and populates this field only if tests will be run.